### ISSUE TITLE
[MULTIPLE BOOKMARK LINES SUPPORT THREAD]

### ISSUE THREAD

**Issue by ao2** [Aris-t2/CustomCSSforFx#99]
# 1. Describe the issue

I have lots of bookmarks in the *bookmark toolbar* section, but only a part of them are shown in the toolbar when enabling `bookmarks_toolbar_multiple_lines.css`

I checked that either scrolling down to the bottom, or setting:
```
  --bookmark_items_height: 999px; /* <- bookmark items - line height */
  --bookmark_items_lines: 1; /* <- maximum amount of lines */
```

only a certain number of bookmarks are shown (95 in my case), not all of them.

# 2. What should have happened instead?

I would have expected to be able to access all bookmarks in the toolbar after scrolling down to the bottom.

# 3. Steps to reproduce the issue?

Just enable `bookmarks_toolbar_multiple_lines.css` and add a lot of bookmarks to the bookmarks toolbar, after a certain point new bookmarks are not shown anymore in the toolbar. They can still be accessed in the bookmarks menu.

# 5. System information
- OS & OS version:  Linux/Debian
- Firefox version:  57.0.3
- Firefox theme / lw-theme:  default theme
- This projects settings the issue occurs with: `bookmarks_toolbar_multiple_lines.css`

Thanks,
   Antonio

**Comment by Aris-t2** [comment]
There might be an internal limit for the maximum amount of visible bookmarks inside "bookmarks item box". This limit does not affect popup menus.

In my tests 109 bookmarks are visible (on Win7 with Fx59) and it does not matter how many visible lines I set 2, 10 or 20.

After all this is an experimental feature offering something not meant to be available by default.

**Comment by ao2** [comment]
> There might be an internal limit for the maximum amount of visible bookmarks inside "bookmarks item box". This limit does not affect popup menus.

I tried the same `userChrome.css` in FF55 and the limitation is not there, all bookmarks are visible by scrolling down the toolbar. I'll do some more research and maybe post a bug report to Firefox itself.

**Comment by Aris-t2** [comment]
In my tests Firefox 52-56 was also free of that limitation. Must be something they introduced in Fx57.

**Comment by ao2** [comment]
For now I mentioned the problem in https://bugzilla.mozilla.org/show_bug.cgi?id=1420089

**Comment by ao2** [comment]
The limit on number of items in the bookmarks toolbar has been introduced by https://hg.mozilla.org/mozilla-central/rev/b3a7b5c0618b to fix https://bugzilla.mozilla.org/show_bug.cgi?id=1392081

**Comment by Aris-t2** [comment]
So there is nothing we can do about. There is no hidden preference to workaround this.

**Comment by shogunreaper** [comment]
i just started trying to use css and i can't get this one to work

no matter what i change  

 --bookmark_items_lines: 2;  

to the number of lines stays at 3.

**Comment by Aris-t2** [comment]
If you are using the complete release archive of this project, you have to change variables inside \config\general_variables.css.

The variable has also to be inside the actual CSS file for users, that only pick some files of this project without using the whole package.

**Comment by shogunreaper** [comment]
thanks that worked

is there a way to increase the width of the area used? There's a ton of unused space on the right side near the scrollbar.

**Comment by Aris-t2** [comment]
Post a screenshot.

**Comment by shogunreaper** [comment]
https://i.imgur.com/UaPiCpx.png

looks like it depends on the name of the previous bookmarks, if it's short then it will have a lot of unused space.

**Comment by Marcosdw** [comment]
/* @import "./css/toolbars/bookmarks_toolbar_multiple_lines.css"; /**/ /* <--- EXPERIMENTAL */

does not work on the latest version Nightly 66
__
Sorry I use an online translator French / English

**Comment by Aris-t2** [comment]
Thanks for letting us know. (Multiple tab lines v1 also refuses to work on current Nightly builds.)

**Comment by Speravir** [comment]
Oh, ~~this is mean~~ what a coincedence: I had the idea to test whether a flex or grid layout would help for multiple lines in Fx66, and therefore downloaded a beta version, but faced some other problems more important in the moment to me, so testing didn’t start yet. Now, you come with the same idea, and one advantage of it is I could test the rule: It works great, thank you!

**Comment by Marcosdw** [comment]
Hello,

I test :  @import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx66.css";
Work fine but please add scrollbar as in older versions because truncated with 3 lines 
Thanks
__
Sorry I use an online translator French / English

**Comment by Marcosdw** [comment]
Hello,

I test in Fx66b9 : @import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx66.css";
Work fine but please add scrollbar as in older versions because truncated with 3 lines
Thanks
__
Sorry I use an online translator French / English

**Comment by Aris-t2** [comment]
This was already fixed by Achille. It will be on next update.

**Comment by Marcosdw** [comment]
Ok thank you ;)

**Comment by ArkadiuszMichalski** [comment]
Hello, I have really strange behaviour when use; `@import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx66.css";`

I use last Nightly 68.a01 and only custom CSS from here (without any addons).

Actually set that options in `general_variables.css`:
```
/* for bookmarks_toolbar_multiple_lines.css */
  --bookmark_items_height: 26px !important; /* <- bookmark items - line height */
  --bookmark_items_lines: 3 !important; /* <- maximum amount of lines */
```
and personal setting in `my_userChrome.css`:
```
#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item{
  margin-top: 0px !important;
  margin-bottom: 0px !important;
  padding-right: 2px !important;
}
```

My bookmarks toolbar has 3 line of bookmark. Last line its not full, just half. Problem is when I try opening some url then all items on bookmarks toolbar disappears for short time and then appears. Time from disappears to appears is different (1s or less), and basically I do not know what it depends on (maybe after load URL all items back). Look for that picture:
![bug1](https://user-images.githubusercontent.com/2730894/54866702-d74b6400-4d77-11e9-9a68-719f26268ad9.gif)

It's strange because happend not for every URL, but for some of them (and not every time:). And loading this url from bookmark toolbar or URL bar make the same blinking result. 

I remember that in past I have the same problem when using Multirow Bookmarks Toolbar Plus extension in new FX version, so I jump to Roomy Bookmarks Toolbar extension and problem dissapears. Has anyone met such behavior and knows how to fix this flashing?

@Aris-t2 any solution? Eventually I will try to import the code from Roomy Bookmarks Toolbar to my uc.js script, but I would prefer a solution in pure CSS (if it's possible).

**Comment by Aris-t2** [comment]
Haven't seen this glitch here yet.

**Comment by ArkadiuszMichalski** [comment]
I have this since FX54 or even earlier. It is possible that my set of bookmarks has such a strange behavior, I will try to reproduce it on a clean profile with minimal changes and few tabs (but minimum 2 line). If you would be interested, I would send a link to such a profile, because without it you will never fix it.

**Comment by Aris-t2** [comment]
Sure, share a link with a new test profile uploaded to a share service like Dropbox ( or similar without waiting times, capchas or registrations ;-) ).

Setup Portable Firefox, add the latest files of the project to /chrome/ folder and setup theme, create a bunch of bookmarks..., zip/rar/7z the /profile/ folder and upload it somewhere.

**Comment by Pastant** [comment]
**1. Describe the issue**

I have lots of bookmarks in the bookmark toolbar section, but the third row is not completely visible when enabling bookmarks_toolbar_multiple_lines.css:

![Unbenannt](https://user-images.githubusercontent.com/32750257/61534251-610df400-aa2f-11e9-9a9b-1dd0aafdbc40.jpg)

  --bookmark_items_height: 35px; /* <- bookmark items - line height */
  --bookmark_items_lines: 4; /* <- maximum amount of lines */

doesn't seem to help.

**2. What should have happened instead?**

Show all three rows without scrolling.

**3. Steps to reproduce the issue?**

Just enable bookmarks_toolbar_multiple_lines.css and add a lot of bookmarks to the bookmarks toolbar. The third row will not show completely. And multiple tab bars is not compatible with this tweak: ./css/tabs/tabs_multiple_lines_v3.css

![2](https://user-images.githubusercontent.com/32750257/61534258-68350200-aa2f-11e9-9ea8-e6f10fce1da1.jpg)

**5. System information**

    OS & OS version: Win 7 64 bit
    Firefox version: 68.0.1
    Firefox theme / lw-theme: default theme


Thanks

**Comment by Aris-t2** [comment]
As far as I know  the combination 'tabs not on top' + 'tabs multiple lines' + 'multiple bookmarks lines' never worked properly.

As a workaround you can try 'tabs_below_navigation_toolbar_fx65_v2.css' + 'tabs_multiple_lines_v3.css' + 'bookmarks_toolbar_multiple_lines_fx66.css'.

**Comment by Pastant** [comment]
Thanks for the response. The workaround works almost perfect ...

The combination 'tabs not on top' + 'tabs multiple lines' + 'multiple bookmarks lines' worked with FF ESR 60.0.8 properly:

![23](https://user-images.githubusercontent.com/32750257/61544592-15b31000-aa46-11e9-8cfc-09725c0a5c00.jpg)

**Comment by Aris-t2** [comment]
I meant it stopped working perfectly after Mozillas tab changes in Fx65.

**Comment by ArkadiuszMichalski** [comment]
This not work in last Nightly 74.

**Comment by Aris-t2** [comment]
I can confirm it.

Since it was not my code, we have to wait for a workaround from the original author.

**Comment by topiland** [comment]
@Aris-t2  sems now tabs width interfere bookmarks toolbar width
  --tab_max_width: 2000px !important; in config fixed bookmark tolbar width
Still was 200 even the   --bookmark_items_lines: 10 worked with one bookmark /line

**Comment by topiland** [comment]
This is what mostly working for me with ff74:

/* Firefox Quantum userChrome.css tweaks ************************************************/
/* Github: https://github.com/aris-t2/customcssforfx ************************************/
/****************************************************************************************/


/* NOTE  ********************************************************************************/
/* Variables are set inside '.\config\' folders CSS files, if complete package is used! */

/* import old button/bookmark size code *************************************************/
@import "./../buttons/buttons_on_bookmarks_toolbar_old_size_and_appearance.css";

:root {
  --bookmark_items_height: 26px; /* <- bookmark items - line height */
  --bookmark_items_lines: 3; /* <- maximum amount of lines */
}

#PersonalToolbar {
	  min-height: var(--bookmark_items_height) !important;
	  max-height: calc(var(--bookmark_items_height)*(var(--bookmark_items_lines))) !important;
    }

    #personal-bookmarks,
    #PlacesToolbar > hbox {
        display: block;
    }

    #PlacesToolbarItems {
        display: flex;
        flex-wrap: wrap;
        overflow-x: visible;
        overflow-y: auto;
    }

#PlacesToolbarItems > .scrollbox-innerbox {
  display: flex;
  flex-wrap: wrap; 
  overflow-x: hidden !important;
  overflow-y: visible !important;
  max-height: calc(var(--bookmark_items_height)*(var(--bookmark_items_lines))) !important
}



#personal-bookmarks #PlacesToolbar {
  display: block;
  min-height: 0;
  overflow-x: hidden;
  overflow-y: auto;
  max-height: calc( var(--bookmark_items_lines) * var(--bookmark_items_height) );
}



#personal-bookmarks #PlacesToolbar > hbox > hbox{
  overflow-x: hidden;
  overflow-y: hidden;
}

#personal-bookmarks #PlacesToolbar #PlacesToolbarDropIndicator[collapsed="true"],
#personal-bookmarks #PlacesToolbar #PlacesToolbarDropIndicator{
  display: none;
}

#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item{
  padding-top: 0px !important;
  padding-bottom: 0px !important;
  margin-top: 2px !important;
  margin-bottom: 2px !important;
  vertical-align: middle;
}
/*
#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item:hover:active:not([disabled="true"]),
#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item[open="true"] {
  padding-top: 2px !important;
  padding-bottom: 2px !important;
}
*/
#personal-bookmarks #PlacesToolbar toolbarseparator{
  -moz-appearance: none !important;
  visibility: visible !important;
  display: inline;
  text-shadow: none !important;
  border-left: 3px solid ThreeDShadow !important;
  border-right: 3px solid ThreeDHighlight !important;
  vertical-align: middle;
}

#personal-bookmarks toolbarbutton.bookmark-item[dragover][open]{
  -moz-appearance: toolbarbutton;
}

#navigator-toolbox #PersonalToolbar { 
  max-height: calc( var(--bookmark_items_lines) * var(--bookmark_items_height) ) !important;
}
![image](https://user-images.githubusercontent.com/26923487/72117673-dad3ca00-334e-11ea-920b-7d3ef2b7f15c.png)

**Comment by Aris-t2** [comment]
Thanks for the new code. I will add it on next update.

**Comment by tramp20** [comment]
@Aris-t2 
Thank you for your updates.
Since Fx 74 beta 3 (and with custom_css_for_fx_v3.0.2_v3) I see only one bookmarks toolbar instead of two.
The newest 74 beta5 shows the same behavior.
Sorry I am not able to understand the CSS code :-)

**Comment by Aris-t2** [comment]
There are three settings inside userChrome.css
```css
/* @import "./css/toolbars/bookmarks_toolbar_multiple_lines.css"; /**/ /* <--- EXPERIMENTAL */
/* @import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx66.css"; /**/ /* <--- EXPERIMENTAL */ /*Fx66-73*/
/* @import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx74.css"; /**/ /* <--- EXPERIMENTAL */ /*Fx74+*/
```

Are you using the correct one?

**Comment by tramp20** [comment]
The last one of course, the first two betas of Firefox were ok with css v3.02.

**Comment by Aris-t2** [comment]
`bookmarks_toolbar_multiple_lines_fx74.css` works here on Fx 73, 74 and 75. Maybe you are using a combination of settings, that does not support multiple bookmarks lines or maybe you are talking about something else, because your are referring to "bookmark toolbars" and not lines/rows of bookmarks.

Post screeshots of how your config looks and how it should look.

**Comment by Aris-t2** [comment]
Using `addonbar_extra_bookmarks_toolbar_below_navbar_fx65.css` and moving bookmarks to navigation toolbar offers this result:
![image](https://user-images.githubusercontent.com/6013569/74849003-843bb180-5338-11ea-9f69-7e55af1d1dc6.png)

Using `bookmarks_toolbar_multiple_lines_fx74.css` and keeping bookmarks on bookmarks toolbar offers this result:
![image](https://user-images.githubusercontent.com/6013569/74849180-b64d1380-5338-11ea-95ee-95264de5fe88.png)

You can not combine both settings, it will not work.

**Comment by tramp20** [comment]
Thank you but I could not solve my problem :-(
So I did a new fresh Fx 74.0b5 installation and now all is ok again.

**Comment by tramp20** [comment]
Sorry but after updating Fx 74.0b3 to Fx 74.0b5 the error comes back.

Firefox 74.0b3:
![Fx_Tabs_below_fx740b3](https://user-images.githubusercontent.com/50739311/74947861-c843ba80-53fb-11ea-920f-83c577d12058.jpg)

Firefox 74.0b5:
![Fx_Tabs_below_fx740b5](https://user-images.githubusercontent.com/50739311/74947899-d560a980-53fb-11ea-94ca-d81c58e2d5c1.jpg)

I am using custom_css_for_fx_v3.0.3 and I did not change anything between the update.
It is reproducible (already 3 times).

**Comment by Aris-t2** [comment]
Not sure what to say, `tabs_below_navigation_toolbar_fx72.css` + `bookmarks_toolbar_multiple_lines_fx74.css` works fine here.

![image](https://user-images.githubusercontent.com/6013569/74957381-bcf68c00-5407-11ea-8a4c-b2c3ea2aa002.png)

**Comment by tramp20** [comment]
> 
> 
> works fine here.
> 
>
I found (hopefully) the culprit:
Since some years I imported a my_userchrome.css (from MZine forum).
This file is ofc outdated (I do not understand what it should do 2 or 3 years ago but it helped).
But now with the latest Fx 74 it prevents the latest custom_css from correct working.

[my_userChrome.zip](https://github.com/Aris-t2/CustomCSSforFx/files/4236121/my_userChrome.zip)

**Comment by Aris-t2** [comment]
Your file was an older version of a multiple bookmarks lines script. It broke the "current" one. Just remove it.

**Comment by tramp20** [comment]
Thank you very much :-)

**Comment by netgrafik** [comment]
Where to add the file "userChrome.css"  in the portable Firefox version?

Firefox Developer 74.0b9  Portable version from the website in : 
http://portableappz.blogspot.com/2012/07/mozilla-firefox-150-beta-2-multilingual.html

Thanks

**Comment by Aris-t2** [comment]
FirefoxPortable\Data\profile\chrome

**Comment by c-stoyanov** [comment]
Hey guys, any idea how to fix the bookmarks to not show one under the other? Thanks.
![Screenshot_1](https://user-images.githubusercontent.com/62072366/76444667-f306aa80-63cc-11ea-9bd0-daea1982e0b8.jpg)

**Comment by Aris-t2** [comment]
Use the most recent files of this project.
"bookmarks_toolbar_multiple_lines_fx74.css"

**Comment by Thomas000101** [comment]
I use the file that came with the 3.03V2  Pack
and it only goves me 2 lines on FF 74 with the scrolling bar :(
I tried to edit the .css file and wrote "6", but it doesnt work.
Is there a newer working version? If yes, can you give us the Link please?

**Comment by Aris-t2** [comment]
I can't do anything about the the used theme. Theme devs mostly provide images for about 3-4 top toolbar lines only, because the default ui does not extend that much.

**Comment by Thomas000101** [comment]
ok, thank you anyway for your great work :)

**Comment by c-stoyanov** [comment]
Thanks very much for the feedback, I've managed to get the desired result, but now I'm getting a huge black bar below the tabs bar? Any ideas?
![Screenshot_2](https://user-images.githubusercontent.com/62072366/76557729-c02fe580-64a4-11ea-9c65-8eb2d0d889fe.jpg)

**Comment by Aris-t2** [comment]
Nope, never saw this one.

**Comment by bmazak** [comment]
You guys are great.  Thanks for the fix.  FF74 now has 4 bookmark lines and is scroll-able.  YeHHH!

**Comment by Achille-Grs** [comment]
@Aris-t2 I solved the problem for the blocked gap on the right side of bookmarks toolbar.
The only thing needed is a width value of the resolution screen monitor plus +5px
in bookmark css file. In my case is 1920x1080.
I just wanted you know it!

#personal-bookmarks, #PlacesToolbar > hbox {
-moz-padding-end: 0px !important;
display: block;
width: 1925px !important;
}

**Comment by memedipa** [comment]
I'm a noob and just miss my rows.
What's not right? Did I just cut and paste it wrong?? 

media (-moz-os-version: windows-win7), (-moz-os-version: windows-win8), (-moz-os-version: windows-win10) {
  @media not all and (-moz-windows-classic) {
	#PersonalToolbar toolbarbutton:not(#PlacesChevron),
	#PersonalToolbar #PlacesToolbar toolbarbutton,
	#PlacesToolbar toolbarbutton:not(#PlacesChevron) {
	  -moz-appearance: toolbarbutton !important;
	  background: unset !important;
	  box-shadow: unset !important;
	  padding-left: 3px !important;
	  padding-right: 3px !important;
	  margin-left: 0px !important;
	  margin-right: 0px !important;
	}

	#PersonalToolbar .toolbarbutton-badge-stack,
	#PersonalToolbar toolbarbutton:not(#PlacesChevron) .toolbarbutton-icon {
	  background: unset !important;
	  box-shadow: unset !important;
	}

	#PersonalToolbar toolbarbutton:not(#PlacesChevron) .toolbarbutton-icon {
	  padding-left: 0px !important;
	  padding-right: 0px !important;
	  width: 16px !important;
	}
	#PersonalToolbar .scrollbutton-up .toolbarbutton-icon,
	#PersonalToolbar .scrollbutton-down .toolbarbutton-icon {
	  padding-left: unset !important;
	  padding-right: unset !important;
	  width: unset !important;
	}
  }
}

:root {

  --bookmark_items_height: 26px; /* <- bookmark items - line height */

  --bookmark_items_lines: 3; /* <- maximum amount of lines */

}




#personal-bookmarks {

  display: block;

}




#personal-bookmarks #PlacesToolbar {

  display: block;

  min-height: 0;

  overflow-x: hidden;

  overflow-y: auto;

  max-height: calc( var(--bookmark_items_lines) * var(--bookmark_items_height) );

}




#personal-bookmarks #PlacesToolbar > hbox {

  display: -moz-stack !important;

  left: 0px;

  right: 0px;

  width: 100%;

}




#personal-bookmarks #PlacesToolbar #PlacesToolbarItems {

  overflow-x: visible;

  overflow-y: visible;

}




#personal-bookmarks #PlacesToolbar #PlacesToolbarItems > box {

  display: block;

}




#personal-bookmarks #PlacesToolbar > .bookmark-item{

  visibility: visible !important;

}




#personal-bookmarks #PlacesToolbar .chevron{

  visibility: collapse;

}




#personal-bookmarks #PlacesToolbar > hbox > hbox{

  overflow-x: hidden;

  overflow-y: hidden;

}




#personal-bookmarks #PlacesToolbar #PlacesToolbarDropIndicator[collapsed="true"],

#personal-bookmarks #PlacesToolbar #PlacesToolbarDropIndicator{

  display: none;

}




#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item{

  padding-top: 0px !important;

  padding-bottom: 0px !important;

  margin-top: 2px !important;

  margin-bottom: 2px !important;

  vertical-align: middle;

}

/*

#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item:hover:active:not([disabled="true"]),

#personal-bookmarks #PlacesToolbar toolbarbutton.bookmark-item[open="true"] {

  padding-top: 2px !important;

  padding-bottom: 2px !important;

}

*/

#personal-bookmarks #PlacesToolbar toolbarseparator{

  -moz-appearance: none !important;

  visibility: visible !important;

  display: inline;

  text-shadow: none !important;

  border-left: 3px solid ThreeDShadow !important;

  border-right: 3px solid ThreeDHighlight !important;

  vertical-align: middle;

}




#personal-bookmarks toolbarbutton.bookmark-item[dragover][open]{

  -moz-appearance: toolbarbutton;

}




#navigator-toolbox #PersonalToolbar { 

  max-height: calc( var(--bookmark_items_lines) * var(--bookmark_items_height) ) !important;

**Comment by Aris-t2** [comment]
On Fx74+ test this code only:

https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/css/toolbars/bookmarks_toolbar_multiple_lines_fx74.css

**Comment by bmazak** [comment]
Thanks but it’s working fine.

**Comment by memedipa** [comment]
I copied and pasted that exact code into a css file. Now, when I open the file in text edit, it turns into rtf when I save so I'm having to do an extra step to change it back to css. That's weird but wasn't sure if it mattered.

Anyway, I get one line of bookmarks still.  All I did was cut and paste. Changed nothing. I have disabled all other add-ons except no script, google translate, and google scholar.

**Comment by Aris-t2** [comment]
Use Notepad++ (or similar) to edit CSS files.

If the only custom change you use is this feature and the code is inside your userChrome.css, everything is setup correctly. You still need to have many bookmarks on bookmarks toolbar. This can not be seen in customizing mode.

**Comment by Ghst84** [comment]
Hey, guys. I'm a complete noob. Could someone please explain to me how to use this code? I need the exact steps. I updated my Firefox to version 74, to try this out, but I don't know what to do. In case this doesn't work, I will revert back to version 56. Thanks!

**Comment by Aris-t2** [comment]
Every step how to setup custom CSS files is mentioned on main page.
https://github.com/Aris-t2/CustomCSSforFx

I suggest to start using the complete package and enable/disable feature you need to get basic understanding how everything works.

**Comment by KathrynBassett** [comment]
New computer (Win10), new installation of Firefox (77.0.1), new install of CustomCSSforFx. Imported my bookmarks by synching FF. On existing computer (Win8.1)

**Comment by Aris-t2** [comment]
Make sure you are using the correct settings and the latest release of this project.

`tabs_multiple_lines_fx71.css` works fine even on current Firefox 79 Nightly.

You might need to start over and set everything up again. There are many changes between Firefox releases and there is no guarantee previous setup work at all.

**Comment by KathrynBassett** [comment]
It's not tabs, it's bookmarks_toolbar_multiple_lines. I started over, and I changed userChrome
@import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx74.css"; /**/ /* <--- EXPERIMENTAL */ /*Fx74+   Kathryn edited*/

I still get 
![image](https://user-images.githubusercontent.com/47371171/84096577-20a4c500-a9b7-11ea-903f-b618adf47b72.png)
instead of 3 rows.

**Comment by bmazak** [comment]
Here’s the code I use.  It works with Win 10.  Let me know if it works for you. I assume you put this in your profile at C:\users\Administrator\Mozilla\Firefox\Profiles\xxxxxxprofilename\chome

Now don’t for get to go into about:config and change the setting 

     Open a new tab in Mozilla Firefox.

    Type about:config in the address bar. ...

    Enter the following text in the search box: toolkit.legacyUserProfileCustomizations.stylesheets  or just legacy.

    Set the option toolkit.legacyUserProfileCustomizations.stylesheets to True .

 

This data is from the same place but has be slightly modified for spacing and some other things, good luck.

/* Firefox Quantum userChrome.css tweaks ************************************************/

/* Github: https://github.com/aris-t2/customcssforfx ************************************/

**Comment by KathrynBassett** [comment]
I did the about:config bit, but that made 3 rows of single icons that didn't scroll, so I set it back. Don't know what you meant by "data is from the same place".

**Comment by Aris-t2** [comment]
`bookmarks_toolbar_multiple_lines_fx74.css` works for me too (up to Fx79).

Test on a new browser profile using latest files.

![image](https://user-images.githubusercontent.com/6013569/84181773-b3159900-aa89-11ea-8702-b3676cbf8745.png)

**Comment by bmazak** [comment]
Correction in the path, sorry.

 

Here’s the code I use.  It works with Win 10.  Let me know if it works for you. I assume you put this in your profile at C:\users\Administrator\App Data\Romming\Mozilla\Firefox\Profiles\xxxxxxprofilename\chome

Now don’t for get to go into about:config and change the setting 

     Open a new tab in Mozilla Firefox.

    Type about:config in the address bar. ...

    Enter the following text in the search box: toolkit.legacyUserProfileCustomizations.stylesheets  or just legacy.

    Set the option toolkit.legacyUserProfileCustomizations.stylesheets to True .

 

This data is from the same place but has be slightly modified for spacing and some other things, good luck.

/* Firefox Quantum userChrome.css tweaks ************************************************/

/* Github: https://github.com/aris-t2/customcssforfx ************************************/

**Comment by bmazak** [comment]
Brian Live would like to recall the message, "[Aris-t2/CustomCSSforFx] [MULTIPLE BOOKMARK LINES SUPPORT THREAD] (#99)".

**Comment by bmazak** [comment]
Changed path

 

Here’s the code I use.  It works with Win 10.  Let me know if it works for you. I assume you put this in your profile at C:\users\Administrator\AppData\Roaming\Mozilla\Firefox\Profiles\xxxxxxprofilename\chome  Note: may be different slightly!

Now don’t for get to go into about:config and change the setting 

     Open a new tab in Mozilla Firefox.

    Type about:config in the address bar. ...

    Enter the following text in the search box: toolkit.legacyUserProfileCustomizations.stylesheets  or just legacy.

    Set the option toolkit.legacyUserProfileCustomizations.stylesheets to True .

 

This data is from the same place but has be slightly modified for spacing and some other things, good luck.

/* Firefox Quantum userChrome.css tweaks ************************************************/

/* Github: https://github.com/aris-t2/customcssforfx ************************************/

**Comment by KathrynBassett** [comment]
> Test on a new browser profile using latest files.

How does one make a "new browser profile"? I thought it was all automatic.

**Comment by Aris-t2** [comment]
@bmazak

Please use Githubs comment function instead of replying to this via email to keep the posts clean and readable.

**Comment by KathrynBassett** [comment]
> Test on a new browser profile using latest files.

Putting all the steps in case someone else runs across my problem, and so I can have the steps to follow them on a friend's computer who is having the same problem.

Made new profile. I know I'm using the new one because it's very basic.

Logged in to Firefox so I could sync to get my bookmarks.

Closed FF

Copied the contents of custom_css_for_fx_v3.1.4_v3 to the new chrome folder. 

Removed /* from @import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx74.css" in userChrome.css in chrome folder. (Caution: I had fx66 edited out in one try, which might have been the problem all along).

Opened FF and changed about:config > toolkit.legacyUserProfileCustomizations.stylesheets >to true

Closed and reopened FF and IT WORKS!!!

**Comment by KathrynBassett** [comment]
New question: In bookmarks_toolbar_multiple_lines_fx74.css, I changed the 3 to 6 for number of rows. It will "scroll" 6 rows, but how do I "see" all 6 rows? Do I need to change it someplace else as well?

**Comment by bmazak** [comment]
The CSS code is the data.  I have scrollable 4 lines so try my code once and see if that helps.

**Comment by Aris-t2** [comment]
No there is no list like that.

Main feutures always have been squared tabs, application button in titlebar and tabs below navigation/bookmarks toolbar.

**Comment by KathrynBassett** [comment]
Phooey! I implemented tabs_below_navigation and that breaks the multiple bookmarks, putting it back to one line with the >> at the end of the row. How can I have both?

**Comment by Aris-t2** [comment]
https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/userChrome.css#L341
https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/userChrome.css#L350

**Comment by KathrynBassett** [comment]
I took out the /*  before these two lines
@import "./css/tabs/tabs_below_navigation_toolbar_fx74.css"; /**/ /* Kathryn edited */
@import "./css/toolbars/bookmarks_toolbar_multiple_lines_fx74.css"; /**/ /* <--- EXPERIMENTAL Kathryn edited*/ /*Fx74+*/

Still not working, so what did I do wrong? Also, is there a specific way I should add my name in the notes so it doesn't mess things up? Is that my problem? Notepad doesn't give line numbers, so I can't give you that, but they were in the area of lines 341 and 350 that you referenced. Adding my name is best way for me to find changes quickly.

**Comment by KathrynBassett** [comment]
Started over again, and left out the notes about what I edited, and it's working now. I can only guess I added the Kathryn edited comment in wrong format. But since I have the lines here, I can refer to them in the future.

**Comment by Aris-t2** [comment]
@KathrynBassett 
I suggest using a more professional editor like Notepad++ to edit files. It offers line numbers, code highlighting and tons of other feature default Notepad does not offer.

**Comment by KathrynBassett** [comment]
Thanks for the suggestion. I'd heard of Notepad++ but didn't know why people used it. It's now downloaded, installed, and available from the right click "open with" menu. The line numbers referred to above that I've commented out line, are 360 (tabs below navigation), and 644 (multiple lines).

Last question is actually for a different thread, but still pertinent here: what is the proper way to add a comment to a line - in these cases, adding "Kathryn edited" so I can find things that I changed versus them being default. I'm pretty sure I did it wrong at one point, which is why it "broke". Everything is working now, but I want to go back and add my comments.

I can just add /* Kathryn commented out the line below/ above a particular line. Only downside I **see**, is that it throws off the line numbers when referring to them in a thread. I'd rather understand the quantity of / and * spaces needed to add something to the end of the line. 

And curiosity only, why the comment experimental, even when something is being used?
